src/main/java/io/cryostat/recordings/RecordingTargetHelper.java

```diff
@@ -540,50 +538,68 @@ private IConstrainedMap<EventOptionID> enableAllEvents(JFRConnection connection)
+                                        connection -> {
+                                            Optional<IRecordingDescriptor> desc =
+                                                    getDescriptorByName(connection, recordingName);
+
+                                            desc =
+                                                    desc.stream()
+                                                            .filter(
+                                                                    r ->
+                                                                            r.getState()
+                                                                                    .equals(
+                                                                                            RecordingState
+                                                                                                    .STOPPED))
+                                                            .findFirst();
+                                            if (desc.isPresent()) {
+                                                String name = desc.get().getName();
+                                                HyperlinkedSerializableRecordingDescriptor linked =
+                                                        new HyperlinkedSerializableRecordingDescriptor(
+                                                                desc.get(),
+                                                                webServer
+                                                                        .get()
+                                                                        .getDownloadURL(
+                                                                                connection, name),
+                                                                webServer
+                                                                        .get()
+                                                                        .getReportURL(
+                                                                                connection, name));
+                                                return linked;
+                                            }
+                                            return null;
+                                        });
+                        promise.complete(linkedDesc);
+                    } catch (Exception e) {
+                        promise.fail(e);
+                    }
+                };
+        long task =
+                this.vertx.setTimer(
                         delay + TIMESTAMP_DRIFT_SAFEGUARD,
-                        TimeUnit.MILLISECONDS);
+                        id -> {
+                            vertx.executeBlocking(
+                                    promiseHandler,
+                                    false,
+                                    result -> {
+                                        if (result.succeeded()) {
```

Comment:
I think this should be `result.failed()`.

Reply:
Great catch, thanks.